.github/workflows/build.yml

```diff
@@ -1,102 +1,141 @@
 name: Build Sage Executables
 
 on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
   release:
-    types: [ created ]
+    types: [created]
 
 jobs:
   build_ubuntu:
     name: Build Sage for Linux
     runs-on: ubuntu-latest
     container: ubuntu:14.04 # GLIBC 2.19
-    steps:
-    - name: Install dependencies
-      run: |
-        sudo apt-get update && sudo apt-get upgrade -y
-        sudo apt-get install -y curl build-essential
-
-    - name: Checkout repository
-      uses: actions/checkout@v2
-
-    - name: Get Cargo toolchain
-      uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-
-    - name: Build Sage
-      uses: actions-rs/cargo@v1
-      with:
-        command: build
-        args: --release
-
-    - name: Upload executable to release
-      uses: ./.github/workflows/upload_artifacts
-      with:
-        name: Linux 64-bit
-        token: ${{ secrets.GITHUB_TOKEN }}
+    strategy:
+      matrix:
+        include:
+          - name: Linux 64-bit
+            target: x86_64-unknown-linux-musl
 
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get upgrade -y
+          sudo apt-get install -y curl build-essential
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Get Cargo toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+
+      - name: Build Sage
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --release --target ${{ matrix.target }}
+
+      - name: Build archive
+        shell: bash
+        run: |
+          sage_version=${{ github.ref_name }}
+          staging="sage-$sage_version-${{ matrix.target }}"
+          mkdir -p "$staging"
+          cp "target/${{ matrix.target }}/release/sage" "$staging/"
```

Comment:
Would you also copy `LICENSE.md` and `README.md` into `$staging/` here, and in the other matrix targets?

Reply:
yes, sounds like a good idea, that's they do in ripgrep

https://github.com/BurntSushi/ripgrep/blob/master/.github/workflows/release.yml#L153-L156

I didn't want to touch too much of the CI, it can quickly become a rabbithole... and assume things but totally doable

Reply:
I checked ripgrep to see what they included, that was the inspiration for my comment 😄 
And yes, totally a rabbit hole

Reply:
not tested but that should do it. I will try tomorrow if needed